includes/rest/class-server.php

```diff
@@ -0,0 +1,103 @@
+<?php
+namespace Activitypub\Rest;
+
+use WP_REST_Response;
+use Activitypub\Signature;
+
+/**
+ * ActivityPub Server REST-Class
+ *
+ * @author Django Doucet
+ *
+ * @see https://www.w3.org/TR/activitypub/#security-verification
+ */
+class Server {
+
+	/**
+	 * Initialize the class, registering WordPress hooks
+	 */
+	public static function init() {
+		\add_filter( 'rest_request_before_callbacks', array( self::class, 'authorize_activitypub_requests' ), 10, 3 );
+		\add_action( 'rest_api_init', array( self::class, 'register_routes' ) );
+	}
+
+	/**
+	 * Register routes
+	 */
+	public static function register_routes() {
+		\register_rest_route(
+			'activitypub/1.0',
+			'/application',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( self::class, 'application_actor' ),
+					'permission_callback' => '__return_true',
+				),
+			)
+		);
+	}
+
+	/**
+	 * Render Application actor profile
+	 *
+	 * @return WP_REST_Response
+	 */
+	public static function application_actor() {
+		$json = new \stdClass();
+
+		$json->{'@context'} = \Activitypub\get_context();
+		$json->id = \get_rest_url( null, 'activitypub/1.0/application' );
+		$json->type = 'Application';
+		$json->preferredUsername = wp_parse_url( get_site_url(), PHP_URL_HOST ); // phpcs:ignore WordPress.NamingConventions
+		$json->name = get_bloginfo( 'name' );
+		$json->summary = 'WordPress-ActivityPub application actor';
+		$json->manuallyApprovesFollowers = true; // phpcs:ignore WordPress.NamingConventions
```

Comment:
is this required by mastodon or activitypub? if not, we should remove it, because you should not be able to follow the application user at all.